### PR TITLE
Allow RegfValue.type to fall back to int

### PR DESCRIPTION
**Pull request by JSCU-CNI**
This PR fixes edge cases where `RegfValue.type` cannot be translated to the `RegistryValueType` enum.

### CODE REVIEW

**Comment by Horofic** [comment]
Would you mind adding a test case for this behaviour in `test_regutil`?

Not sure, but should this not return `REG_NONE` in this case? Took a cursory peek here https://learn.microsoft.com/en-us/windows/win32/sysinfo/registry-value-types.

**Comment by Schamper** [comment]
Using `_missing_` is probably nicer: https://docs.python.org/3/library/enum.html#enum.Enum._missing_

File: dissect/target/helpers/regutil.py [hunk 1]
```diff
@@ -707,7 +724,7 @@ def value(self) -> ValueType:
         return self.kv.value
 
     @property
-    def type(self) -> RegistryValueType:
+    def type(self) -> RegistryValueType | int:
```
Comment (Schamper) [inline dissect/target/helpers/regutil.py:727]:
```suggestion
    def type(self) -> RegistryValueType:
```
No longer necessary.

**Review by Schamper — CHANGES_REQUESTED** [review]

**Comment by Schamper** [comment]
> Added in [308fda1](https://github.com/fox-it/dissect.target/pull/956/commits/308fda1cb84d9ac2dd63087bb8af0032664740f8). Perhaps the Enum should be moved to the `dissect.regf` project?

Maybe, I'm fine with it in here for now since this enum serves more as an abstraction layer than a "parsing" enum.

**Review by Schamper — APPROVED** [review]

**Review by Schamper — APPROVED** [review]
